<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="ARPGL000231" name="BasicTests.LayerArchitecture" tests="1" failures="1" timestamp="19 Jul 2021 06:56:26 GMT" time="0.010" errors="0">
  <testcase name="layer_architecture" time="0.010" classname="BasicTests.LayerArchitecture">
    <failure type="java.lang.AssertionError" message="Architecture Violation [Priority: MEDIUM] - Rule &#039;Layered architecture consisting of
layer &#039;controller&#039; (&#039;..controller..&#039;)
layer &#039;service&#039; (&#039;..service..&#039;)
layer &#039;entity&#039; (&#039;..entity..&#039;)
where layer &#039;controller&#039; may not be accessed by any layer
where layer &#039;service&#039; may only be accessed by layers [&#039;controller&#039;]&#039; was violated (1 times):
Field &lt;demo.entity.Customer.obj&gt; has type &lt;demo.controller.CustomerController&gt; in (Customer.java:0)">
      <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'Layered architecture consisting of
layer 'controller' ('..controller..')
layer 'service' ('..service..')
layer 'entity' ('..entity..')
where layer 'controller' may not be accessed by any layer
where layer 'service' may only be accessed by layers ['controller']' was violated (1 times):
Field <demo.entity.Customer.obj> has type <demo.controller.CustomerController> in (Customer.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.library.Architectures$LayeredArchitecture.check(Architectures.java:291)
	at BasicTests.LayerArchitecture.layer_architecture(LayerArchitecture.java:24)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at BasicTests.ExcelGenerate.main(ExcelGenerate.java:25)
]]>
    </failure>
  </testcase> <!-- layer_architecture -->
</testsuite> <!-- BasicTests.LayerArchitecture -->
